<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Wei Liu" />

<meta name="date" content="2025-10-03" />

<title>DR-SC: simulation</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">DR-SC: simulation</h1>
<h4 class="author">Wei Liu</h4>
<h4 class="date">2025-10-03</h4>



<div id="generate-the-simulated-data" class="section level2">
<h2>Generate the simulated data</h2>
<p>First, we analyze a simulated spatial transcriptomics data from ST
platform, which is a Seurat object format. It is noted that the
<code>meta.data</code> must include spatial coordinates in columns named
“row” (x coordinates) and “col” (y coordinates)! This data is available
<a href="https://github.com/feiyoung/DR-SC.Analysis/tree/main/data/">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;sim_seu.rds&#39;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(DR.SC)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">head</span>(seu<span class="sc">@</span>meta.data)</span></code></pre></div>
</div>
<div id="fit-dr-sc-using-simulated-data" class="section level2">
<h2>Fit DR-SC using simulated data</h2>
<div id="data-preprocessing" class="section level3">
<h3>Data preprocessing</h3>
<p>This preprocessing includes Log-normalization and feature selection.
Here we select highly variable genes for example first. The selected
genes’ names are saved in “<a href="mailto:seu@assays" class="email">seu@assays</a>$<a href="mailto:RNA@var.features" class="email">RNA@var.features</a>”.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="do">### Given K</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>seu <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(seu)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co"># choose highly variable features using Seurat</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>seu <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(seu, <span class="at">nfeatures =</span> <span class="dv">400</span>)</span></code></pre></div>
</div>
<div id="fit-dr-sc-based-on-highly-variable-geneshvgs" class="section level3">
<h3>Fit DR-SC based on highly variable genes(HVGs)</h3>
<p>For function <code>DR.SC</code>, users can specify the number of
clusters <span class="math inline">\(K\)</span> or set <code>K</code> to
be an integer vector by using modified BIC(MBIC) to determine <span class="math inline">\(K\)</span>. First, we try using user-specified
number of clusters. Then we show the version chosen by MBIC.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="do">### Given K</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>seu2 <span class="ot">&lt;-</span> <span class="fu">DR.SC</span>(seu, <span class="at">q=</span><span class="dv">15</span>, <span class="at">K=</span><span class="dv">4</span>, <span class="at">platform =</span> <span class="st">&#39;ST&#39;</span>,  <span class="at">verbose=</span>F, <span class="at">approxPCA=</span>T)</span></code></pre></div>
<p>After finishing model fitting, we use ajusted rand index
(<code>ARI</code>) to check the performance of clustering</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>mclust<span class="sc">::</span><span class="fu">adjustedRandIndex</span>(seu2<span class="sc">$</span>spatial.drsc.cluster, seu<span class="sc">$</span>true_clusters)</span></code></pre></div>
<p>Next, we show the application of DR-SC in visualization. First, we
can visualize the clusters from DR-SC on the spatial coordinates.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">spatialPlotClusters</span>(seu2)</span></code></pre></div>
<p>We can also visualize the clusters from DR-SC on the two-dimensional
tSNE based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">drscPlot</span>(seu2)</span></code></pre></div>
<p>Show the UMAP plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">drscPlot</span>(seu2, <span class="at">visu.method =</span> <span class="st">&#39;UMAP&#39;</span>)</span></code></pre></div>
<p>Use MBIC to choose number of clusters:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>seu2 <span class="ot">&lt;-</span> <span class="fu">DR.SC</span>(seu, <span class="at">q=</span><span class="dv">10</span>, <span class="at">K=</span><span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>, <span class="at">platform =</span> <span class="st">&#39;ST&#39;</span>, <span class="at">verbose=</span>F,<span class="at">approxPCA=</span>T)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">mbicPlot</span>(seu2)</span></code></pre></div>
</div>
<div id="fit-dr-sc-based-on-spatially-variable-genessvgs" class="section level3">
<h3>Fit DR-SC based on spatially variable genes(SVGs)</h3>
<p>First, we select the spatilly variable genes using funciton
<code>FindSVGs</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="do">### Given K</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>seu <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(seu, <span class="at">verbose=</span>F)</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co"># choose 400 spatially variable features using FindSVGs</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>seus <span class="ot">&lt;-</span> <span class="fu">FindSVGs</span>(seu, <span class="at">nfeatures =</span> <span class="dv">400</span>, <span class="at">verbose =</span> F)</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>seu2 <span class="ot">&lt;-</span> <span class="fu">DR.SC</span>(seus, <span class="at">q=</span><span class="dv">15</span>, <span class="at">K=</span><span class="dv">4</span>, <span class="at">platform =</span> <span class="st">&#39;ST&#39;</span>, <span class="at">verbose=</span>F)</span></code></pre></div>
<p>Using ARI to check the performance of clustering</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>mclust<span class="sc">::</span><span class="fu">adjustedRandIndex</span>(seu2<span class="sc">$</span>spatial.drsc.cluster, seu<span class="sc">$</span>true_clusters)</span></code></pre></div>
</div>
<div id="dr-sc-can-enhance-visualization" class="section level3">
<h3>DR-SC can enhance visualization</h3>
<p>Show the spatial scatter plot for clusters</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">spatialPlotClusters</span>(seu2)</span></code></pre></div>
<p>Show the tSNE plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">drscPlot</span>(seu2)</span></code></pre></div>
<p>Show the UMAP plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">drscPlot</span>(seu2, <span class="at">visu.method =</span> <span class="st">&#39;UMAP&#39;</span>)</span></code></pre></div>
</div>
<div id="dr-sc-can-automatically-determine-the-number-of-clusters" class="section level3">
<h3>DR-SC can automatically determine the number of clusters</h3>
<p>Use MBIC to choose number of clusters:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>seu2 <span class="ot">&lt;-</span> <span class="fu">DR.SC</span>(seus, <span class="at">q=</span><span class="dv">4</span>, <span class="at">K=</span><span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>, <span class="at">platform =</span> <span class="st">&#39;ST&#39;</span>,  <span class="at">verbose=</span>F)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="fu">mbicPlot</span>(seu2)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="co"># or plot BIC or AIC</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="co"># mbicPlot(seu2, criteria = &#39;BIC&#39;)</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co"># mbicPlot(seu2, criteria = &#39;AIC&#39;)</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a><span class="co"># tune pen.const</span></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>seu2 <span class="ot">&lt;-</span> <span class="fu">selectModel</span>(seu2, <span class="at">pen.const =</span> <span class="fl">0.7</span>)</span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a><span class="fu">mbicPlot</span>(seu2)</span></code></pre></div>
</div>
</div>
<div id="dr-sc-can-help-differentially-expression-analysis" class="section level2">
<h2>DR-SC can help differentially expression analysis</h2>
<p>Conduct visualization of marker gene expression. ### Ridge plots
Visualize single cell expression distributions in each cluster from
Seruat.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(seu2)</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(dplyr) )</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a><span class="co"># Find the top 1 marker genes, user can change n to access more marker genes</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>dat <span class="sc">%&gt;%</span><span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>    <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">1</span>, <span class="at">wt =</span> avg_log2FC) <span class="ot">-&gt;</span> top</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>genes <span class="ot">&lt;-</span> top<span class="sc">$</span>gene</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a><span class="fu">RidgePlot</span>(seu2, <span class="at">features =</span> genes, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
<div id="violin-plot" class="section level3">
<h3>Violin plot</h3>
<p>Visualize single cell expression distributions in each cluster</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">VlnPlot</span>(seu2, <span class="at">features =</span> genes, <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="feature-plot" class="section level3">
<h3>Feature plot</h3>
<p>We extract tSNE based on the features from DR-SC and then visualize
feature expression in the low-dimensional space</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>seu2 <span class="ot">&lt;-</span> <span class="fu">RunTSNE</span>(seu2, <span class="at">reduction=</span><span class="st">&quot;dr-sc&quot;</span>, <span class="at">reduction.key=</span><span class="st">&#39;drsctSNE_&#39;</span>)</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seu2, <span class="at">features =</span> genes, <span class="at">reduction =</span> <span class="st">&#39;tsne&#39;</span> ,<span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="dot-plots" class="section level3">
<h3>Dot plots</h3>
<p>The size of the dot corresponds to the percentage of cells expressing
the feature in each cluster. The color represents the average expression
level</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">DotPlot</span>(seu2, <span class="at">features =</span> genes)</span></code></pre></div>
</div>
<div id="heatmap-plot" class="section level3">
<h3>Heatmap plot</h3>
<p>Single cell heatmap of feature expression</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co"># standard scaling (no regression)</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>dat <span class="sc">%&gt;%</span><span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>    <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">wt =</span> avg_log2FC) <span class="ot">-&gt;</span> top</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="do">### select the marker genes that are also the variable genes.</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(top<span class="sc">$</span>gene, <span class="fu">VariableFeatures</span>(seu2))</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a><span class="do">## Change the HVGs to SVGs</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="co">#  &lt;- topSVGs(seu2, 400)</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>seu2 <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seu2, <span class="at">verbose =</span> F)</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a><span class="fu">DoHeatmap</span>(<span class="fu">subset</span>(seu2, <span class="at">downsample =</span> <span class="dv">500</span>),<span class="at">features =</span> genes, <span class="at">size =</span> <span class="dv">5</span>)</span></code></pre></div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
